File: invoice-agent-system-java/src/main/java/com/example/invoice/agent/KoogAgentRuntime.java
package com.example.invoice.agent;

import org.springframework.stereotype.Component;

/**
 * Koog integration seam.
 *
 * The build includes ai.koog:koog-agents:1.0.0. The Spring Boot 4.0.7 application is Java 21 based,
 * while Koog can be introduced behind InvoiceAgent<I, O> implementations for tool-enabled agents,
 * memory, tracing, and agent lifecycle management.
 */
@Component
public class KoogAgentRuntime {
    public String agentRuntimeName() {
        return "Koog 1.0.0 runtime seam for Java 21 invoice agents";
    }
}

File: invoice-agent-system/src/main/kotlin/com/example/invoice/agent/KoogAgentRuntime.kt
package com.example.invoice.agent

import org.springframework.stereotype.Component

/**
 * Koog integration seam.
 *
 * The build includes ai.koog:koog-agents:1.0.0. Keep the rest of the Spring application
 * behind the InvoiceAgent<I, O> contract so each processing step can be backed by:
 *  - a Koog AIAgent with tools,
 *  - a direct Spring AI ChatClient call,
 *  - deterministic business logic, or
 *  - a human review workbench adapter.
 *
 * In production, implement each InvoiceAgent with a Koog tool-enabled agent and keep this
 * class as the factory/registry for agent lifecycle, tracing, persistence, and memory.
 */
@Component
class KoogAgentRuntime {
    fun agentRuntimeName(): String = "Koog 1.0.0 runtime seam for invoice agents"
}
